### PR TITLE
feat: default table name populated in query editor for mock datasources

### LINKED ISSUE #25247: [Feature]: Default queries for mock datasources [Users]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

This is a follow-up to https://github.com/appsmithorg/appsmith/issues/23960. 

With the above story, we pick the first available entity (table/collection, etc) and show the user a `read` query by default. With a [[real datasource](https://www.notion.so/7b5b3f6302cc4a8e93f8ecd1a3965cad?pvs=21)](https://www.notion.so/7b5b3f6302cc4a8e93f8ecd1a3965cad?pvs=21), it is not possible to know which is the most meaningful entity to choose. However, with mock datasources, we do have the option because we own the underlying databases.

****************Users db****************

1. When the user adds a query on the `Users` mock db, the query should be
    1. Query: `SELECT * FROM public.users LIMIT 10;` 
    2. Use prepared statement: `On`
    
Screenshot 2023-07-04 at 12 09 15

    

### Why should this be worked on?

When I add a mock query, I should see a usable query on default, so that I get impressed by appsmith’s power to easily build functional apps

### PR DESCRIPTION
**Pull request by sneha122**
## Description

This is a follow-up to https://github.com/appsmithorg/appsmith/issues/23960.

In above issue, we picked the first available entity (table/collection, etc) and show the user a read query by default. With a [[real datasource](https://www.notion.so/7b5b3f6302cc4a8e93f8ecd1a3965cad?pvs=21)](https://www.notion.so/7b5b3f6302cc4a8e93f8ecd1a3965cad?pvs=21), it is not possible to know which is the most meaningful entity to choose. However, with mock datasources, we do have the option because we own the underlying databases.

This PR adds that functionality, In case of Users mock DB, we will populate the query editor with `select * from public.users limit 10`, as Users mock DB contains users table.

#### PR fixes following issue(s)
Fixes #25247 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
#### Type of change
- New feature (non-breaking change which adds functionality)
>
>
>
- [x] Manual
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [x] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress changes looks ok!

File: app/client/src/utils/editorContextUtils.ts [hunk 1]
```diff
@@ -171,3 +174,22 @@ export function getFormDiffPaths(initialValues: any, currentValues: any) {
...
+ * Returns mock datasource default table name to be populated in query editor, based on plugin name
+ * @param pluginId string
+ * @returns string
+ */
+export function getSQLPluginsMockTableName(pluginId: string) {
+  const state: AppState = store.getState();
+  const plugin: Plugin | undefined = getPlugin(state, pluginId);
+
+  switch (plugin?.name) {
+    case PluginName.POSTGRES: {
```
Comment (AmanAgarwal041) [inline app/client/src/utils/editorContextUtils.ts:188]: Are we not handling for mongo mock ds ?

**Review by AmanAgarwal041 — APPROVED** [review]

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress changes are ok to proceed.

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**